Tests/SwiftParserTest/AvailabilityTests.swift

```diff
@@ -99,4 +99,93 @@ final class AvailabilityTests: XCTestCase {
       """
     )
   }
+
+  func testVersionParsing() {
+    AssertParse(
+      """
+      @available(OSX 10)
+      func test() {}
+      """,
+      substructure: Syntax(
+        VersionTupleSyntax(
+          major: .integerLiteral("10")
+        )
+      )
+    )
+
+    AssertParse(
+      """
+      @available(OSX 10.0)
+      func test() {}
+      """,
+      substructure: Syntax(
+        VersionTupleSyntax(
+          major: .integerLiteral("10"),
+          minorPeriod: .periodToken(),
+          minor: .integerLiteral("0")
+        )
+      )
+    )
+
+    AssertParse(
+      """
+      @available(OSX 10.0.1)
+      func test() {}
+      """,
+      substructure: Syntax(
+        VersionTupleSyntax(
+          major: .integerLiteral("10"),
+          minorPeriod: .periodToken(),
+          minor: .integerLiteral("0"),
+          patchPeriod: .periodToken(),
+          patchVersion: .integerLiteral("1")
+        )
+      )
+    )
+
```

Comment:
Worth having custom diagnostics for these? It would be nice if the non-digit-containing token was unexpected within the version tuple rather than ending up as unexpected within the attribute when there's no space after `.`.

Reply:
My judgment was that the diagnostics were sufficiently good for now.